### PR TITLE
[tanium] Initial Release for the Tanium

### PR DESCRIPTION
**Pull request by vinit-elastic**
## What does this PR do?

- Generated the skeleton of the Tanium integration package.
- Added data streams.
- Added data collection logic for all the data streams.
- Added the ingest pipeline for all the data streams.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yml files.
- Added dashboards and visualizations.
- Added test for pipeline for all the data streams.
- Added system test cases for all the data streams.

~~**Note**: This integration supports 6 data streams. Out of Which, the `Threat Response` data stream is developed and tested with sample data. However, the other 5 data streams are developed and tested against the live data.~~

## Integration release checklist
This checklist is intended for integrations maintainers to ensure consistency
when creating or updating a Package, Module or Dataset for an Integration.

### All changes
- [x] Change follows the [contributing guidelines](https://github.com/elastic/integrations/blob/master/CONTRIBUTING.md)
- [x] Supported versions of the monitoring target is documented
- [x] Supported operating systems are documented (if applicable)
- [x] Integration or [System tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/system_testing.md) exist
- [x] Documentation exists
- [x] Fields follow [ECS](https://github.com/elastic/ecs) and [naming conventions](https://www.elastic.co/guide/en/beats/devguide/master/event-conventions.html)
- [x] At least a manual test with ES / Kibana / Agent has been performed.
- [x] Required Kibana version set to: ^7.17.0 || ^8.0.0

### New Package
- [x] Screenshot of the "Add Integration" page on Fleet added

### Dashboards changes
- [x] Dashboards exists
- [x] Screenshots added or updated
- [x] Datastream filters added to visualizations

### Log dataset changes
- [x] [Pipeline tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/pipeline_testing.md) exist (if applicable)
- [x] Generated output for at least 1 log file exists
- [x] Sample event (sample_event.json) exists

## How to test this PR locally

- Clone integrations repo.
- Install elastic package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/tanium directory.
- Run the following command to run tests. 
> elastic-package test 

## Screenshots

image (12)
image (13)
image (11)
image (9)
image (10)

### CODE REVIEW

**Comment by syedrafice** [comment]
Is there a targeted release version for this integration?

**Comment by vinit-elastic** [comment]
Hey @syedrafice - This integration will be available on Elastic stack versions greater than 7.17. and 8.x.

**Comment by syedrafice** [comment]
Thank you @vinit-elastic. I should have clarified, what release will the integration get merged into the main integrations GitHub repo for the public to use?

**Comment by jamiehynds** [comment]
Hey @syedrafice - we're currently going through the PR review process and expect the Tanium integration to ship over the coming weeks. We don't have to wait for a stack release. 

Do you have a customer looking for the integration? We're shipping the integration as beta, and will be looking for early feedback if you have a user in mind.

**Comment by P1llus** [comment]
As discussed on the weekly meeting, feel free to add the dynamic template as well, according to: https://github.com/elastic/integrations/issues/5055

File: packages/tanium/data_stream/threat_response/_dev/test/pipeline/test-threat-response.log-expected.json [hunk 1]
```diff
@@ -0,0 +1,7675 @@
...
+                                        "bits": 64,
+                                        "os": {
+                                            "platform": "\"Ubuntu",
+                                            "value": "\"Ubuntu 18.04.6 LTS\"",
+                                            "version": "18.04.6 LTS\""
+                                        },
+                                        "platform": "linux"
+                                    },
+                                    "threat_id": "850394582,983962193,2667062006,3398416878,3999014465",
+                                    "whats": [
```
Comment (P1llus) [inline packages/tanium/data_stream/threat_response/_dev/test/pipeline/test-threat-response.log-expected.json:145]: Looking at fields.yml, this is mapped as a normal group, and not something like nested/arrays of objects, is this correct?

**Review by P1llus — APPROVED** [review]
From what I can see its ready, but its quite a heavy integration, especially the threat datastream. A bit unsure how much we can see from a simple review.

I will leave it here for a little to see if anyone else have more comments.

File: packages/tanium/data_stream/threat_response/_dev/test/pipeline/test-threat-response.log-expected.json (hunk 1 above)
Comment (vinit-elastic) [reply packages/tanium/data_stream/threat_response/_dev/test/pipeline/test-threat-response.log-expected.json:145]: Ah, I see. It should be nested. We will update it.

**Review by P1llus — APPROVED** [review]

**Comment by P1llus** [comment]
Since this includes proper testing, and has been tested against a live instance of the actual product, I will decide to merge.

**Pull request merged**